services/system/Accounts/ui/src/components/AccountSelection.tsx

```diff
@@ -0,0 +1,722 @@
+                      : `Import account`}
+                  </Button>
+                </form>
+              </Form>
+            </DialogHeader>
+          </DialogContent>
+          <div className="max-w-lg mx-auto">
+            <div className="text-center text-muted-foreground py-2">
+              <span>
+                {isInitialLoading
+                  ? "Loading..."
+                  : isNoAccounts
+                  ? `No accounts available.`
+                  : isInvite
+                  ? `Select an account to accept invite to `
+                  : `Select an account to login to `}
+                {!isNoAccounts && (
+                  <span className="text-primary">{appName}</span>
+                )}
+              </span>
+            </div>
+            {!isNoAccounts && (
+              <div className="relative ml-auto flex-1 md:grow-0 mb-3">
+                <Search className="absolute left-2.5 top-2.5 h-4 w-4 text-muted-foreground" />
+                <Input
+                  value={activeSearch}
+                  type="search"
+                  onChange={(e) => setActiveSearch(e.target.value)}
+                  placeholder="Search..."
+                  className="w-full rounded-lg bg-background pl-8 "
+                />
+              </div>
+            )}
+
+            <div className="flex flex-col gap-3 ">
+              <div className="flex flex-col gap-3 max-h-[600px] overflow-auto">
+                {isAccountsLoading
+                  ? [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14].map(
+                      (_, index) => (
+                        <Account
+                          onSelected={() => {}}
+                          isSelected={false}
+                          key={index}
+                        />
+                      )
+                    )
+                  : accountsToRender.map((account) => (
+                      <Account
+                        onSelected={() => {
+                          setSelectedAccountId(account.id);
+                        }}
+                        isSelected={selectedAccountId == account.id}
+                        key={account.id}
+                        name={account.account}
+                      />
+                    ))}
+              </div>
+              {isInvite && !isInviteClaimed && (
+                <button
+                  onClick={() => {
+                    setIsCreatingAccount(true);
+                    setIsModalOpen(true);
+                  }}
+                  className="flex p-4 shrink-0 items-center justify-center rounded-md border border-neutral-600 text-muted-foreground hover:text-primary hover:underline border-dashed"
+                >
+                  Create a new account
+                </button>
+              )}
+            </div>
+            {!isNoAccounts && (
+              <div className="my-3">
+                <Button
+                  disabled={!selectedAccount || isTxInProgress}
```

Comment:
When creating an account, the accept invite button could be disabled if the new account name is invalid or taken. 
Otherwise (or possibly also), you need to add exception handling to onSubmit to handle the promise rejection when acceptWithNewAccount is called with an invalid name

Reply:
Button now disables, expects an account to be available when accepting an invite and expects it to be taken when doing an import.